data/crac/crac-util/src/main/java/com/farao_community/farao/data/crac_util/CracAliasesCreator.java

```diff
@@ -32,13 +33,42 @@ public void createAliases(Crac crac, Network network) {
 
     public void createAliases(Crac crac, Network network, UcteNodeMatchingRule rule) {
         // List (without duplicates) all the crac elements that need to be found in the network
+        Map<String, BranchCnec> aliasedCnecs = new HashMap<>();
         Set<String> elementIds = new HashSet<>();
-        crac.getBranchCnecs().forEach(cnec -> elementIds.add(cnec.getNetworkElement().getId()));
+        crac.getBranchCnecs().forEach(cnec -> aliasedCnecs.put(cnec.getNetworkElement().getId(), cnec));
         crac.getContingencies().forEach(contingency -> handleAliases(contingency.getNetworkElements(), elementIds));
         crac.getNetworkActions().forEach(networkAction -> handleAliases(networkAction.getNetworkElements(), elementIds));
         crac.getRangeActions().forEach(rangeAction -> handleAliases(rangeAction.getNetworkElements(), elementIds));
 
         // Try to find a corresponding element in the network, and add elementId as an alias
+        // Cnecs
+        Map<String, Identifiable> reversedAliases = new HashMap<>();
+        aliasedCnecs.forEach((id, cnec) -> {
+            Optional<Identifiable<?>> correspondingElement = network.getIdentifiables().stream().filter(identifiable -> anyMatchDirect(identifiable, id, rule)).findAny();
+            if (correspondingElement.isPresent()) {
+                correspondingElement.get().addAlias(id);
+            } else {
+                Optional<Identifiable<?>> reversedMatch = network.getIdentifiables().stream().filter(identifiable -> anyMatchReversed(identifiable, id, rule)).findAny();
+                reversedMatch.ifPresent(identifiable -> reversedAliases.put(id, identifiable));
+                cnec.getThresholds().forEach(branchThreshold -> {
+                    Optional<Double> min = branchThreshold.min();
+                    Optional<Double> max = branchThreshold.max();
+                    if (min.isPresent()) {
+                        branchThreshold.max() = Optional.of(-min.get());
+                    } else {
+                        branchThreshold.max() = Optional.of(Double.MAX_VALUE);
+                    }
+                    if (max.isPresent()) {
+                        branchThreshold.min() = Optional.of(-max.get());
+                    } else {
+                        branchThreshold.min() = Optional.of(-Double.MAX_VALUE);
```

Comment:
This doesn't work, what can I do?
(the pb is that branchThreshold.max() is a value, not a variable, so I can't write value = something else)

Reply:
the max() method is a getter, so you cannot change the max() value.

Actually there is no setter for the min/max values of a Threshold. I don't really know if its intentional (it ensures that the threshold values cannot be changed, so it could be), or if it's just that we didn't have this need until now.

The easiest way to to this would be to add the setMin() and setMax() methods in the Threshold API and its implementations. Then you would be able to reverse the min and max values.

But i could be interesting to have also the opinion of TheMaskedTurtle who recently worked on the threshold --> maybe he intentionally decided to keep the min/max values finals.

Reply:
We are facing very similar problems in CORE data-sets (inversions of from/to of the Cnecs + inversions of thresholds of the Cnecs). We decided to handle them "outside of farao", in a pre-processing script (which is not in an open-source repository).

Maybe you could directly use this script (it also handles some problems which exists with the transformers and tie-lines, that you might face in the future :s) -> I can help you use it.

Moreover, we are planning a refactoring of the Crac importers which will notably solve that kind of problems ! :) The idea would be that the importer takes a Crac **AND** a Network. And that all the mismatches between the Crac and the Network are dealt with during the import, and not in some wonky steps afterward.

Our first intention is to do it for CORE data, so with flowbased-document and UCTE network. --> so if you are using the same format, we will soon have something better for this...

By "soon", I mean 1-2 months :angel: